### PR TITLE
DROOLS-3958 Unexpected Conflicting rows on rules analysis using date-effective and date-expires on Guided Decision Tables

### PR DESCRIPTION
**Pull request by Rikkola**
https://issues.jboss.org/browse/DROOLS-3958

@jomarko @manstis You two might be the best to review. Thank you.

PRs:
https://github.com/kiegroup/drools/pull/2324
https://github.com/kiegroup/drools-wb/pull/1131

### CODE REVIEW

**Review by jomarko — APPROVED** [review]
Just one comment otherwise looks good. Thank you!

File: drools-verifier/drools-verifier-core/src/test/java/org/drools/verifier/core/index/model/ActivationTimeTest.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                {new ActivationTime(null, null), new ActivationTime(null, new Date(110)), true},
+                {new ActivationTime(null, new Date(110)), new ActivationTime(null, null), true},
+                {new ActivationTime(null, null), new ActivationTime(new Date(100), null), true},
+                {new ActivationTime(new Date(100), null), new ActivationTime(null, null), true},
+        });
+    }
+
+    @Test
+    public void equalDates() {
+        assertEquals(expected, start.overlaps(end));
```
Comment (jomarko) [inline drools-verifier/drools-verifier-core/src/test/java/org/drools/verifier/core/index/model/ActivationTimeTest.java:73]: I think we compare overlay of dates not their equality, could we please choose more appropriate test method name?

Comment (Rikkola) [reply drools-verifier/drools-verifier-core/src/test/java/org/drools/verifier/core/index/model/ActivationTimeTest.java:73]: Good point

File: drools-verifier/drools-verifier-core/src/main/java/org/drools/verifier/core/index/model/DateEffectiveRuleAttribute.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
+package org.drools.verifier.core.index.model;
```
Comment (manstis) [inline drools-verifier/drools-verifier-core/src/main/java/org/drools/verifier/core/index/model/DateEffectiveRuleAttribute.java:16]: License please.

File: drools-verifier/drools-verifier-core/src/main/java/org/drools/verifier/core/index/model/DateExpiresRuleAttribute.java [hunk 3]
```diff
@@ -0,0 +1,32 @@
+package org.drools.verifier.core.index.model;
```
Comment (manstis) [inline drools-verifier/drools-verifier-core/src/main/java/org/drools/verifier/core/index/model/DateExpiresRuleAttribute.java:16]: License please.

File: drools-verifier/drools-verifier-core/src/main/java/org/drools/verifier/core/index/model/DateEffectiveRuleAttribute.java (hunk 2 above)
Comment (Rikkola) [reply drools-verifier/drools-verifier-core/src/main/java/org/drools/verifier/core/index/model/DateEffectiveRuleAttribute.java:16]: 2019 and Michael still catches my missing licenses. You would imagine I would learn to add them 🌝

**Review by manstis — APPROVED** [review]

**Comment by Rikkola** [comment]
@kiegroup/gatekeepers Please merge the bundle.

**Pull request merged**